server/service/conditional_access_microsoft_proxy/conditional_access_microsoft_proxy.go

```diff
@@ -0,0 +1,267 @@
+	TenantID string `json:"entra_tenant_id"`
+	Secret   string `json:"fleet_server_secret"`
+}
+
+func (p *Proxy) Create(ctx context.Context, tenantID string) (*CreateResponse, error) {
+	var createResponse CreateResponse
+	if err := p.post(
+		"/api/v1/microsoft-compliance-partner",
+		createRequest{TenantID: tenantID},
+		&createResponse,
+	); err != nil {
+		return nil, fmt.Errorf("create integration failed: %w", err)
+	}
+	return &createResponse, nil
+}
+
+type GetResponse struct {
+	TenantID        string  `json:"entra_tenant_id"`
+	SetupDone       bool    `json:"setup_done"`
+	AdminConsented  bool    `json:"admin_consented"`
+	AdminConsentURL string  `json:"admin_consent_url"`
+	SetupError      *string `json:"setup_error"`
+}
+
+func (p *Proxy) Get(ctx context.Context, tenantID string, secret string) (*GetResponse, error) {
+	var getResponse GetResponse
+	if err := p.get(
+		"/api/v1/microsoft-compliance-partner/settings",
+		fmt.Sprintf("entraTenantId=%s&fleetServerSecret=%s", tenantID, secret),
+		&getResponse,
+	); err != nil {
+		return nil, fmt.Errorf("get integration settings failed: %w", err)
+	}
+	return &getResponse, nil
+}
+
+type DeleteResponse struct {
+	Error string `json:"error"`
+}
+
+func (p *Proxy) Delete(ctx context.Context, tenantID string, secret string) (*DeleteResponse, error) {
+	var deleteResponse DeleteResponse
+	if err := p.delete(
+		"/api/v1/microsoft-compliance-partner/settings",
+		fmt.Sprintf("entraTenantId=%s&fleetServerSecret=%s", tenantID, secret),
+		&deleteResponse,
+	); err != nil {
+		return nil, fmt.Errorf("delete integration failed: %w", err)
+	}
+	return &deleteResponse, nil
+}
+
+type setComplianceStatusRequest struct {
+	TenantID        string `json:"entraTenantId"`
+	Secret          string `json:"fleetServerSecret"`
+	DeviceID        string `json:"deviceId"`
+	DeviceName      string `json:"deviceName"`
+	OS              string `json:"os"`
+	OSVersion       string `json:"osVersion"`
+	UserID          string `json:"userId"`
+	Compliant       bool   `json:"compliant"`
+	LastCheckInTime int    `json:"lastCheckInTime"`
+}
+type SetComplianceStatusResponse struct {
+	MessageID string `json:"message_id"`
+}
+
+func (p *Proxy) SetComplianceStatus(
+	ctx context.Context,
+	tenantID string, secret string,
+	deviceID, deviceName, osName, osVersion string,
+	compliant bool,
+	lastCheckInTime time.Time,
+) (*SetComplianceStatusResponse, error) {
+	var setComplianceStatusResponse SetComplianceStatusResponse
+	if err := p.post(
+		"/api/v1/microsoft-compliance-partner/device",
+		setComplianceStatusRequest{
+			TenantID:        tenantID,
+			Secret:          secret,
+			DeviceID:        deviceID,
+			DeviceName:      deviceName,
+			OS:              osName,
+			OSVersion:       osVersion,
+			Compliant:       compliant,
+			LastCheckInTime: int(lastCheckInTime.Unix()),
+
+			// TODO(lucas): We currently don't know how to get this.
+			// We will find out when integrating with the test tenant.
+			UserID: "TODO",
```

Comment:
This is okay being merged in like this?

Reply:
Yes, we've agreed to. 

Feature can only be used if the `FLEET_MICROSOFT_COMPLIANCE_PARTNER_PROXY_URI` and  `FLEET_MICROSOFT_COMPLIANCE_PARTNER_PROXY_API_KEY` environment variables are set.